File: Tier1/HttpClients/ClientInterfaces/AgreementInterface.cs
using Shared.Domain;
using Shared.DTOs;

namespace HttpClients.ClientInterfaces;

public interface AgreementInterface
{
    Task<RequestAgreementDTO> RequestAgreement(RequestAgreementDTO dto);
    Task<RespondAgreementDTO> RespondToAgreement(RespondAgreementDTO dto);
    Task<AgreementsByHostDTO> GetAllRequestsByHostDTO(AgreementsByHostDTO dto);
}

File: Program/Tier2/src/main/java/via/sep3/group11/tier2/logicInterfaces/AgreementInterface.java
package via.sep3.group11.tier2.logicInterfaces;

import via.sep3.group11.tier2.shared.DTOs.*;
import via.sep3.group11.tier2.shared.domain.Agreement;

import java.util.List;

public interface AgreementInterface {
    AgreementDTO requestAgreement(RequestAgreementDTO dto);
    AgreementDTO respondToAgreement(RespondAgreementDTO dto);
    AgreementListDTO getAllRequestsByHost(AgreementsByHostDTO dto);
    AgreementDTO getAgreementByRefugeeEmail(String refugeeEmail);
}
